flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1486,22 +1486,17 @@ private void handleTimerException(Exception ex) {
 	}
 
 	private ProcessingTimeCallback deferCallbackToMailbox(MailboxExecutor mailboxExecutor, ProcessingTimeCallback callback) {
-		return timestamp -> {
-			mailboxExecutor.execute(
-				() -> invokeProcessingTimeCallback(callback, timestamp),
-				"Timer callback for %s @ %d",
-				callback,
-				timestamp);
-		};
+		return timestamp ->
+				mailboxExecutor.execute(() -> {
+							try {
```

Comment:
Complicated lambdas are in general [discouraged](https://dzone.com/articles/functional-programming-patterns-with-java-8).

Reply:
Not relevant anymore, removed, thanks.

Not sure though, what's more complicated :)